src/base64.mo

```diff
@@ -0,0 +1,43 @@
+import Result "mo:base/Result";
+import Text "mo:base/Text";
+
+module {
+  func stripAnyPrefix(str : Text, prefixes : [Text]) : ?Text {
+    for (p in prefixes.vals()) {
+      if (Text.startsWith(str, #text p)) {
+        return Text.stripStart(str, #text p);
+      };
+    };
+    return null;
+  };
+
+  public func validateImage(imageStr : Text) : Result.Result<(), Text> {
+    let ?data = stripAnyPrefix(
+      imageStr,
+      [
+        "data:image/png;base64,",
+        "data:image/svg+xml;base64,",
+        "data:image/jpeg;base64,",
+      ],
+    ) else return #err("Logo image must be in PNG, JPEG, or SVG format");
+
+    let byteSize = (Text.size(data) * 3) / 4;
+    if (byteSize > 32_768) {
+      return #err("Logo image must be no larger than 32 kB");
+    };
+
+    if (Text.contains(data, #predicate isNotBase64)) {
+      return #err("Logo image is corrupted");
+    };
+
+    return #ok();
+  };
+
+  func isBase64(c : Char) : Bool {
+    (c >= 'A' and c <= 'Z') or (c >= 'a' and c <= 'z') or (c >= '0' and c <= '9') or c == '+' or c == '/' or c == '=';
+  };
+
+  func isNotBase64(c : Char) : Bool {
```

Comment:
Does it make sense to duplicate the logic for a negative check if we can use `not isBase64(char)`?

Reply:
For our application it is not needed. I only wanted to have the code for both conditions (positive and negative) written down and checked in so that others can copy-paste the code if they need it.

If someone writes a performance critical application, for example for very large files, then they will want to avoid to wrap the function call. Also the negative test has 7 comparisons vs 9 for the positive test. So they will prefer the direct negative test.

Reply:
Got it, thanks